File: api/src/main/java/gg/mineral/api/world/property/Dimension.java
package gg.mineral.api.world.property;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public enum Dimension {
    NETHER((byte) -1), OVERWORLD((byte) 0), END((byte) 1);

    @Getter
    private final byte id;
}

File: api/src/main/java/gg/mineral/api/world/property/Dimension.kt
package gg.mineral.api.world.property

import lombok.Getter
import lombok.RequiredArgsConstructor

@RequiredArgsConstructor
enum class Dimension {
    NETHER(-1.toByte()), OVERWORLD(0.toByte()), END(1.toByte());

    @Getter
    private val id: Byte = 0
}
